### ISSUE TITLE
JavaScript TypeError breaks capital loans page 

### ISSUE THREAD

**Issue by darcie** [Automattic/woocommerce-payments#7531]
### Describe the bug

Visiting Payments > Capital shows an error and does not load the expected Capital information.

```
There was an error rendering this view. Please contact support for assistance if the problem persists.
TypeError: Cannot read properties of undefined (reading 'toString')
```
Reported with 6.6.1 with PHP 7.4.33 and 6.6.2 with PHP 8.2.11. No reports of the same error before yesterday. 

From the site logs, the capital information is successfully retrieved from the account but not displayed on the page. 

### To Reproduce

1. Go to Payments > Capital on a site with a capital loan
2. See error displayed

### Additional context

7192208-zen
7195743-zen

**Comment by csmcneill** [comment]
I tried testing this out locally using the dev tools and I am unable to. Enabling Capital doesn't yield a **Payments > Capital** menu item, but I've also never really tested out the Capital flow before today.

- I enabled Capital for `acct_1O3StBCGaUlzmonG`—no luck.
- I added a test financing offer with the status of `accepted`—no luck.
- I later changed the status to `paid_out`—no luck.
- I added a couple other test offers—no luck.
- I reverted from 6.6.2 to 6.6.0—no luck.
- I reverted from 6.6.0 to 5.3.0—no luck.

**Comment by csmcneill** [comment]
I tried to accept a test loan, but it hangs when I attempt to add an email.

I'm getting some console errors while in the Stripe dashboard. Here's an excerpt:

```
       POST https://api.stripe.com/v1/capital/offer_applications/cptloffrapp_1O3TDWCGaUlzmonGS3oY68e2?include_only%5B%5D=id%2Cobject%2Ccreated%2Cmerchant%2Cselected_offer%2Cselected_principal_amount%2Cannual_revenue%2Ccurrency%2Creview_deadline%2Cattestations.attested_at%2Cattestations.attested_by%2Cattestations.id%2Cfinancing_intended_use%2Ctotal_cash_equivalents%2Coutstanding_obligations 500 (Internal Server Error)
```

The `500` error makes me suspect that there might be an issue on the Stripe side of things.

**Comment by jessy-p** [comment]
@tpaksu Would your team be working on this area?

**Comment by tpaksu** [comment]
Hey @jessy-p, I think with the latest changes to the responsibilities, the feature seems to be owned by Helix now, according to Pc2DNy-3z-p2.

**Comment by jessy-p** [comment]
Thanks @tpaksu !
Assigning to Helix (based on team responsibilities Pc2DNy-3z-p2) @haszari. Assigning as part of Gamma Triage process PcreKM-yM-p2.

**Comment by haszari** [comment]
Labelling as `critical` since the error breaks the whole capital UI. However it's a relatively low priority (i.e. capital loans are a lower priority than things like taking payments :) 

Adding to our current maintenance queue, hopefully we can get this in WooPayments 6.7.

**Comment by haszari** [comment]
Next steps:

- Work out how to reproduce the issue. Thanks @csmcneill for investigating 🙌
- Fix!

**Comment by haszari** [comment]
@darcie @csmcneill (and any HEs seeing reports) – please add as much detail as possible about the error (if merchants have access to it!). If we can get a screenshot of the error in browser dev tools, recent / related network traffic that might give us a few more clues. 

- What line / file is this error happening on.
- Network requests related to capital loans.
- Other network requests that power that page.

Thanks!

**Comment by haszari** [comment]
I haven't found a way to reproduce this on my dev site. My dev site does have a test loan (i.e. a fake loan taken out in test mode). The page renders correctly as far as I can tell.

<img width="1385" alt="Screenshot 2023-10-27 at 9 38 43 AM" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/2071387b-dd30-472a-90c7-34988a1e4daf">

---

However, I am seeing a different JS issue (likely unrelated) when I view `Payments > Capital Loans`. 

```
Warning: Received NaN for the `max` attribute. If this is expected, cast the value to a string.
    at input
    at label
    at div
    at div
    at Bc (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/components/index.js?ver=1695595385:2:630536)
    at div
    at http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:18358:47
    at UnconnectedFlex (http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:26116:7)
    at UnconnectedCardFooter (http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:42244:23)
    at div
    at http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:18358:47
    at div
    at http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:18358:47
    at BaseContextSystemProvider (http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:15797:3)
    at UnconnectedCard (http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:41727:7)
    at Zd (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/components/index.js?ver=1695595385:2:691462)
    at ErrorBoundary (webpack-internal:///./client/components/error-boundary/index.js:19:5)
    at div
    at Page (webpack-internal:///./client/components/page/index.tsx:27:5)
    at CapitalPage (webpack-internal:///./client/capital/index.tsx:164:60)
    at Suspense
    at Z (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1695595385:2:451865)
    at div
    at div
    at Fr (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1695595385:2:537012)
    at div
    at SlotFillProvider (http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:21134:3)
    at provider_SlotFillProvider (http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:21157:5)
    at Provider (http://localhost:8082/wp-includes/js/dist/components.js?ver=32407b70de56366bfd74:21302:3)
    at w (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/admin-layout/index.js?ver=1695595385:1:2144)
    at Wr (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1695595385:2:537580)
    at zr (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1695595385:2:537361)
    at http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1695595385:2:540445
    at http://localhost:8082/wp-includes/js/dist/data.js?ver=16121b61f8bc66f02ac6:4132:22
    at WithSelect(Component) (http://localhost:8082/wp-includes/js/dist/compose.js?ver=6861d91c4896ce3aecda:3117:10)
    at http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/data/index.js?ver=1695595385:2:126650
    at N (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1695595385:2:418319)
    at M (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1695595385:2:417760)
    at a (http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1695595385:2:412291)
    at http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1695595385:2:540698
    at http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/data/index.js?ver=1695595385:2:127732
    at http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/data/index.js?ver=1695595385:2:124602
    at http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/data/index.js?ver=1695595385:2:124602
    at http://localhost:8082/wp-content/plugins/woocommerce/assets/client/admin/data/index.js?ver=1695595385:2:125824
```

**Comment by haszari** [comment]
> @darcie @csmcneill (and any HEs seeing reports) – please add as much detail as possible about the error (if merchants have access to it!). If we can get a screenshot of the error in browser dev tools, recent / related network traffic that might give us a few more clues. 

~~Also … do we have info about whether these merchants have active loans? More info about loan status might help refine impact and how to reproduce. For example, is the issue is affecting merchants with no loan (just looking at the page?) or merchants with an active loan in a specific state (e.g. fully repaid, partially repaid, awaiting first instalment, etc).~~

> Go to Payments > Capital on a site with a capital loan

Doh, just saw this 🤦🏻

**Comment by darcie** [comment]
They do both have active loans, I can see `financing_paydown` repayment transactions on the account overview pages. 

The error appears very simply: 
<img width="1476" alt="Screenshot 2023-10-26 at 1 37 16 PM" src="https://github.com/Automattic/woocommerce-payments/assets/318285/83c6dcc0-3f34-4f8b-a993-99e7fbffac99">

There is a console error: 
<img width="1028" alt="Screenshot 2023-10-26 at 1 39 30 PM" src="https://github.com/Automattic/woocommerce-payments/assets/318285/b0a594f9-22d1-4371-a2a3-466947aafe86">

In the Network tab, `/wp-json/wc/v3/payments/capital/active_loan_summary?_locale=user` is getting a response. It's just not displayed on the page.

**Comment by haszari** [comment]
> 7192208-zd-a8c
> 7195743-zd-a8c

These two sites are on 6.6.1 and 6.6.2. So it's possible this is a regression in 6.6.1, or 6.6, or earlier. @darcie @csmcneill if it's possible for merchants to try downgrading that might help isolate problematic code.

**Comment by haszari** [comment]
> 7192208-zd-a8c
7195743-zd-a8c

Both sites are on WooCommerce core 8.2.1, so that might be relevant. If it's possible to test with 8.2 that would be useful info too.

**Comment by haszari** [comment]
Clutching at straws … I searched the changes in [6.6.0](https://github.com/Automattic/woocommerce-payments/commit/137b8be2f10b54fb9759f142ae87d5e9f8bc84e0#) and [6.6.1](https://github.com/Automattic/woocommerce-payments/commit/7ee6541838ed472f67e0438991f10cc4de3019b3) for `toString`, looking for risky code. Didn't find anything, only this use of URLSearchParams for WooPay code, which I believe should be fine, and seems unlikely to fire anywhere near capital loans.

https://github.com/Automattic/woocommerce-payments/blob/137b8be2f10b54fb9759f142ae87d5e9f8bc84e0/client/checkout/woopay/express-button/woopay-express-checkout-button.js#L160C1-L167

```js
const urlParams = new URLSearchParams();
urlParams.append( 'testMode', getConfig( 'testMode' ) );
urlParams.append( 'source_url', window.location.href );
urlParams.append( 'tracksUserIdentity', tracksUserId );

// Low-chance hypothesis: urlParams is undefined
return (
  getConfig( 'woopayHost' ) + '/connect/?' + urlParams.toString()
);
```

There is a change in backend, adding a hook for capital loans request (?), might be worth verifying this is not related.

https://github.com/Automattic/woocommerce-payments/commit/137b8be2f10b54fb9759f142ae87d5e9f8bc84e0#diff-24dd496c539f7b016a9956d5e99078c945f6f6ed7648f37e5a4cd1fbd4ba551e

```php
// Low risk hypothesis: the loan data is not getting returned to client because of some problem with the new `assign_hook` approach
public function get_active_loan_summary() {
	$request = Request::get( WC_Payments_API_Client::CAPITAL_API . '/active_loan_summary' );
	// return $request->handle_rest_request( 'wcpay_get_active_loan_summary_request' );
	$request->assign_hook( 'wcpay_get_active_loan_summary_request' ); // NEW!
	return $request->handle_rest_request();
}

public function get_loans() {
	$request = Request::get( WC_Payments_API_Client::CAPITAL_API . '/loans' );
	// return $request->handle_rest_request( 'wcpay_get_loans_request' );
	$request->assign_hook( 'wcpay_get_loans_request' ); // NEW!
	return $request->handle_rest_request();
}
```

**Comment by haszari** [comment]
Tagging in @brucealdridge  to investigate and try to reproduce. Bruce can you timebox your investigation appropriately ? i.e. enough time to thoroughly investigate, but if you hit a brick wall, comment here and we can revisit when we have more info.

If we can't reproduce this we may not be able to fix; though possibly we can find the problem via code inspection.

**Comment by tpaksu** [comment]
I opened the Capital page on my local setup and saw the same error, the error seems to be related with the pagination component (I didn't pull the latest develop, and there were 41 commits after my local develop): 

<img width="1167" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3295/efc4a25a-6d16-4a7a-90f9-43efb8481a21">

Pulled the latest and rebuilt the client code, refreshed the page, and the same error still exists. 

Ran `npm ci` and rebuilt the client code again, still the same. 

Dug a bit deeper, and saw that the other tables are using a property named `rowsPerPage` property, which doesn't exist on the Capital table, tried adding that to the the `TableCard` component on `capital/index.tsx` with a value of `10`, and it got fixed.
